nntrainer/layers/rnn.cpp

```diff
@@ -54,7 +109,49 @@ void RNNLayer::setProperty(const PropertyType type, const std::string &value) {
 }
 
 void RNNLayer::forwarding(bool training) {
-  // NYI
+  Tensor &weight_xh =
+    weightAt(static_cast<int>(RNNParams::weight_xh)).getVariableRef();
+  Tensor &weight_hh =
+    weightAt(static_cast<int>(RNNParams::weight_hh)).getVariableRef();
+  Tensor &bias_h =
+    weightAt(static_cast<int>(RNNParams::bias_h)).getVariableRef();
+
+  Tensor &hidden_ = net_hidden[0]->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+
+  Tensor temp;
+  Tensor hs_prev;
+  Tensor hs;
+
+  for (unsigned int b = 0; b < input_dim[0].batch(); ++b) {
```

Comment:
Please add a TODO here to try and transpose by the batch dimension and see if its faster that this implementation.

Reply:
done.